lib/auth.js

```diff
@@ -6,12 +6,18 @@ var CookieJar = require('cookiejar');
 /**
  * SwaggerAuthorizations applys the correct authorization to an operation being executed
  */
-var SwaggerAuthorizations = module.exports.SwaggerAuthorizations = function () {
-  this.authz = {};
+var SwaggerAuthorizations = module.exports.SwaggerAuthorizations = function (authz) {
```

Comment:
I really don't think we need to keep this unless it's for backward compatibility.  `SwaggerAuthorizations` is really just a JavaScript object from an implementation perspective so wrapping it doesn't make much sense.

Of course, how the review of the rest of this PR goes depends on whether we agree on this or not.